### PR TITLE
Add POW and SQRT

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2357

### CODE REVIEW

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -843,6 +843,51 @@ private void ExecuteInstruction()
...
+                        var exponent = (int)Pop().GetInteger();
+                        var value = Pop().GetInteger();
+                        Push(BigInteger.Pow(value, exponent));
+                        break;
+                    }
+                case OpCode.SQRT:
+                    {
+                        var value = Pop().GetInteger();
+
+                        if (value < 9)
```
Comment (erikzhang) [inline src/neo-vm/ExecutionEngine.cs:857]: Why is this condition needed?

Comment (shargon) [reply src/neo-vm/ExecutionEngine.cs:857]: SpeedUp, I can remove it

Comment (shargon) [reply src/neo-vm/ExecutionEngine.cs:857]: mmm.. The result it's different, maybe we need a different formula

**Review by erikzhang — APPROVED** [review]

File: src/neo-vm/Utility.cs [hunk 2]
```diff
@@ -12,5 +14,22 @@ static Utility()
             StrictUTF8.DecoderFallback = DecoderFallback.ExceptionFallback;
             StrictUTF8.EncoderFallback = EncoderFallback.ExceptionFallback;
         }
+
+        public static BigInteger Sqrt(this BigInteger value)
+        {
+            if (value < 0) throw new InvalidOperationException("value can not be negative");
+            if (value.IsZero) return BigInteger.Zero;
+            if (value < 4) return BigInteger.One;
```
Comment (vncoelho) [inline src/neo-vm/Utility.cs:22]: Maybe it should be $0$ as well, no?

Comment (vncoelho) [reply src/neo-vm/Utility.cs:22]: That is true, it should be One! I just thought about it again.

**Pull request merged**